tests/agent/agent_test.py

```diff
@@ -316,3 +329,135 @@ def on_max_cyclic_process_reached(self, message: agent_message.Message) -> None:
     test_agent.process_message(f"v3.report.vulnerability.{uuid.uuid4()}", message.raw)
 
     assert process_mock.called is False
+
+
+def testProcessMessage_whenExceptionRaised_shouldLogErrorWithMessageAndSystemLoad(
+    agent_run_mock: agent_testing.AgentRunInstance,
+    mocker: plugin.MockerFixture,
+) -> None:
+    """When an exception is raised, the agent should log the exception, the message that was processed,
+    and details about the current state of the system such as cpu, ram..."""
+
+    # Prepare
+    logger_error = mocker.patch("logging.Logger.error")
+
+    class TestAgent(agent.Agent):
+        """Helper class to test OpenTelemetry mixin implementation."""
+
+        def process(self, message: agent_message.Message) -> None:
+            raise ValueError("some error")
+
+    agent_definition = agent_definitions.AgentDefinition(
+        name="agentX",
+        out_selectors=["v3.report.vulnerability"],
+    )
+    agent_settings = runtime_definitions.AgentSettings.from_proto(
+        runtime_definitions.AgentSettings(
+            key="some_key",
+        ).to_raw_proto()
+    )
+    test_agent = TestAgent(
+        agent_definition=agent_definition, agent_settings=agent_settings
+    )
+    actual_message = agent_message.Message.from_data(
+        "v3.healthcheck.ping",
+        {
+            "body": "Hello, can you hear me?",
+        },
+    )
+    control_message = agent_message.Message.from_data(
+        "v3.control",
+        {
+            "control": {"agents": ["agentY", "agentX", "agentX", "agentX"]},
+            "message": actual_message.raw,
+        },
+    )
+
+    # Act
+    test_agent.process_message(
+        f"v3.healthcheck.ping.{uuid.uuid4()}", control_message.raw
+    )
+
+    # Assert
+    assert logger_error.call_count == 3
+    # Assert  System information is logged
+    assert "System Load: %s" in logger_error.call_args_list[0][0][0]
+    assert (
+        isinstance(
+            logger_error.call_args_list[0][0][1], system_info_helper.SystemLoadInfo
+        )
+        is True
+    )
+    # Assert the message is logged
```

Comment:
Remove comments that are useless, please.

Reply:
Why they are useless?